File: src/Python/easy/sorting/problem414.py
from typing import List

class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        nums = list(set(nums))
        nums.sort(reverse=True)
        return nums[2] if len(nums) >= 3 else nums[0]



# nums = [2,2,3,1]
nums = [-1,2,3]

solution = Solution()
solution.thirdMax(nums)

File: src/JavaScript/easy/sorting/problem414.js
/**
 * @param {number[]} nums
 * @return {number}
 */

var thirdMax = function(nums) {
    nums = Array.from(new Set(nums.sort((a,b) => b - a))) 
    return nums.length >= 3 ? nums[2] : nums[0]
};


// let nums = [3,2,1]
// let nums = [1,2]
let nums = [2,2,3,1]

thirdMax(nums)